### PR TITLE
Launch flask app for verifying plots

### PR DESCRIPTION
**Pull request by sophiaandaloro**
nest_local_flask contains three new python scripts:
1. interaction_keys.py to make it easier to specify interaction types of interest (plan to move to init or a different location where all nestpy users can access it.)
2. benchmark_plots.py which defines functions necessary to calculate yields of basic interactions and then plot them.
3. main.py which, when run, launches a Flask app that displays the plots called in benchmark_plots easily via a browser.

The motivation is to get this Flask app-type setup to where we can use it to cross-check nestpy (and thus new NEST updates) automatically via the Flask app.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+.DS_Store
+
+nest_local_flask/__pycache__/benchmark_plots.cpython-37.pyc
```
Comment (tunnell) [inline .gitignore]: Is this needed?

File: nest_local_flask/benchmark_plots.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.axes as axes
+import matplotlib.pylab as pylab
+
+import nestpy 
+from interaction_keys import GetInteractionObject
+
+v_nestpy = str(nestpy.__version__)
```
Comment (tunnell) [inline nest_local_flask/benchmark_plots.py]: already string, also don't need variable

File: nest_local_flask/benchmark_plots.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+v_nest = '2.0.1'
```
Comment (tunnell) [inline nest_local_flask/benchmark_plots.py]: nestpy.__nest_version__, same thing

File: nest_local_flask/benchmark_plots.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
+import numpy as np
```
Comment (tunnell) [inline nest_local_flask/benchmark_plots.py]: Please add docstring at the top saying what this file does.

"""Makes Flask nest plots

More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info More info 
"""

File: nest_local_flask/benchmark_plots.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+detector = nestpy.DetectorExample_XENON10()
+# Performing NEST calculations according to the given detector example       
+nc = nestpy.NESTcalc(detector)
+#Once you have interaction, you can get yields
+#Begin with np.vectorize so that we can get yields over a range of interaction types/energies/drift fields.
+
+@np.vectorize
+def GetYieldsVectorized(interaction, yield_type, **kwargs):
+    # This function does nc.GetYields for the various interactions and arguments we pass into it
+    # TODO: Look at docstrings
```
Comment (tunnell) [inline nest_local_flask/benchmark_plots.py]: docstrings :)

File: nest_local_flask/benchmark_plots.py [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+            return np.nan
+        if interaction_object == GetInteractionObject('gammaray') and kwargs['energy'] > 3e3:
+            return np.nan
+        if interaction_object == GetInteractionObject('beta') and kwargs['energy'] > 3e3:
+            return np.nan     
+    yield_object = nc.GetYields(interaction = interaction_object, **kwargs)
+    # returns the yields for the type of yield we are considering 
+    return getattr(yield_object, yield_type)
+
+def PhotonYield(**kwargs):
```
Comment (tunnell) [inline nest_local_flask/benchmark_plots.py]: docstring

File: nest_local_flask/benchmark_plots.py [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+    return GetYieldsVectorized(yield_type = 'PhotonYield', **kwargs)
+ 
+def ElectronYield(**kwargs):
```
Comment (tunnell) [inline nest_local_flask/benchmark_plots.py]: docstring

File: nest_local_flask/benchmark_plots.py [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+    return GetYieldsVectorized(yield_type = 'ElectronYield', **kwargs)
+
+def Yield(**kwargs):
```
Comment (tunnell) [inline nest_local_flask/benchmark_plots.py]: docstring

Comment (tunnell) [reply nest_local_flask/benchmark_plots.py]: https://en.wikipedia.org/wiki/Docstring#Python

File: nest_local_flask/benchmark_plots.py [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+    return {'photon' : PhotonYield(**kwargs),
+            'electron' : ElectronYield(**kwargs),
+           # What is missing?  Aren't there other parts of YieldObject?
+           }
+
+#Calculate yields for various benchmark calculations
```
Comment (tunnell) [inline nest_local_flask/benchmark_plots.py]: `kwargs = {'energy': energies.T, 'drift_field': fields}`

Then replace below with:

`ElectronYield(interaction='nr', **kwargs)/energies`

Could energies be done somewhere else for less repetition?  Lots of repetition here.

File: nest_local_flask/benchmark_plots.py [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+beta_electrons = ElectronYield(interaction='beta', energy=energies.T, drift_field = fields).T/energies
+beta_photons = PhotonYield(interaction='beta', energy=energies.T, drift_field = fields).T/energies
+gamma_electrons = ElectronYield(interaction='gammaray', energy=energies.T, drift_field = fields).T/energies
+gamma_photons = PhotonYield(interaction='gammaray', energy=energies.T, drift_field = fields).T/energies
+alpha_electrons = ElectronYield(interaction='ion', Z=2, A = 4, energy=energies.T, drift_field = fields).T/energies
+alpha_photons = PhotonYield(interaction='ion',Z=2, A=4, energy=energies.T, drift_field = fields).T/energies
+Pb_electrons = ElectronYield(interaction='ion', Z=82, A = 206, energy=energies.T, drift_field = fields).T/energies
+Pb_photons = PhotonYield(interaction='ion',Z=82, A=206, energy=energies.T, drift_field = fields).T/energies
+
+# ## Nuclear Recoils Plotting
```
Comment (tunnell) [inline nest_local_flask/benchmark_plots.py]: this is docstring, put in

File: nest_local_flask/interaction_keys.py [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import nestpy
+
+#GetInteractionObject grabs the number for the interaction you want so you don't have to always reference the dictionary. Just type e.g., 'ion'
+#It just changes the name to a number for nestpy to do its work.
+def GetInteractionObject(name):
```
Comment (tunnell) [inline nest_local_flask/interaction_keys.py]: Can this be moved into src/nestpy/__init__py, or a file there that __init__py imports?

File: nest_local_flask/main.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
+import io
```
Comment (tunnell) [inline nest_local_flask/main.py]: docstring at top :)

**Comment by tunnell** [comment]
I'd expand the pull request text too so people know

File: nest_local_flask/interaction_keys.py (hunk 11 above)
Comment (sophiaandaloro) [reply nest_local_flask/interaction_keys.py]: Should I do this in the same commit or wait and get the files up and running with docstrings first, then afterwards, work on changing a source file?

**Comment by tunnell** [comment]
can you resolve conflicts then can merge into next patch?

**Comment by sophiaandaloro** [comment]
Closed due to the benchmark plots having their own repository independent of the source code. 
see: github.com/NESTcollaboration/benchmark_plots

**Pull request closed without merging**